tests/integration/specific_executor_params/test_specific_params.py

```diff
@@ -32,3 +33,65 @@ def process(self, docs, parameters, **kwargs):
         )
 
         assert docs[0].tags['assert']
+
+
+def test_specific_params_with_branched_flow():
+    class TextEncoderTestSpecific(Executor):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.model = lambda t: np.random.rand(
+                len(t), 128
+            )  # initialize dummy text embedding model
+
+        @requests(on='/encode')
+        def encode_text(self, docs, parameters, **kwargs):
+            path = parameters.get('access_path', None)
+            text_docs = docs[path]
+            embeddings = self.model(text_docs[:, 'text'])
+            text_docs.embeddings = embeddings
+
+    class EmbeddingCombinerTestSpecific(Executor):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.model = lambda emb1, emb2: np.concatenate(
+                [emb1, emb2], axis=1
+            )  # initialize dummy model to combine embeddings
+
+        @requests(on='/encode')
+        def combine(self, docs, parameters, **kwargs):
+            text1_path = parameters.get('text1_access_path', None)
```

Comment:
Can we have asserts directly in the Executors to make sure that the params arrive correctly?

Reply:
if they Get None the Executor will raise exception so this is also checked.